pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -393,76 +394,116 @@ func (r *rbac) createResources(ctx context.Context) error {
+			logger.Infow("Ensures Default SA in ", "Namespace", ns.GetName())
+			sa, err := r.ensureSA(ctx, &ns)
 			if err != nil {
-				return fmt.Errorf("failed to re-fetch namespace %s: %v", ns.Name, err)
+				withError = true
+				logger.Errorf("failed to ensure default SA in namespace %s, %v", ns.Name, err)
 			}
 
-			// Get the current labels, or initialize if none exist
-			nsLabels := updatedNS.GetLabels()
-			if nsLabels == nil {
-				nsLabels = map[string]string{}
+			if sa != nil {
+				// We use a namespace scoped Role when SCC annotation is present, and
+				// a cluster scoped ClusterRole when the default SCC is used
+				roleRef := r.getSCCRoleInNamespace(&ns)
+				if roleRef != nil {
+					if err := r.ensurePipelinesSCCRoleBinding(ctx, sa, roleRef); err != nil {
+						withError = true
+						logger.Errorf("failed to create Pipeline Scc Role Binding in namespace %s, %v", ns.Name, err)
+					}
+				}
+				if err := r.ensureRoleBindings(ctx, sa); err != nil {
+					withError = true
+					logger.Errorf("failed to create rolebinding in namespace %s, %v", ns.Name, err)
+				}
+
+				// Locking section to ensure clusterRoleBinding is updated by one goroutine at a time
+				mu.Lock()
+				if err := r.ensureClusterRoleBindings(ctx, sa); err != nil {
+					withError = true
+					logger.Errorf("failed to create clusterrolebinding in namespace %s, %v", ns.Name, err)
+				}
+				mu.Unlock()
 			}
 
-			// Add `openshift-pipelines.tekton.dev/namespace-reconcile-version` label to namespace
-			// so that rbac won't loop on it again
-			nsLabels[namespaceVersionLabel] = r.version
-			updatedNS.SetLabels(nsLabels)
+			if !withError {
+				logger.Infof("add label namespace-reconcile-version to mark namespace '%s' as reconciled", ns.Name)
+
+				// Re-fetch the namespace to ensure we have the latest version
+				updatedNS, err := r.kubeClientSet.CoreV1().Namespaces().Get(ctx, ns.Name, metav1.GetOptions{})
+				if err != nil {
+					mu.Lock()
+					unreconciledNamespaces = append(unreconciledNamespaces, ns.Name)
+					mu.Unlock()
+					return
+				}
+
+				// Get the current labels, or initialize if none exist
+				nsLabels := updatedNS.GetLabels()
+				if nsLabels == nil {
+					nsLabels = map[string]string{}
+				}
 
-			// Update the namespace with set labels
-			if _, err = r.kubeClientSet.CoreV1().Namespaces().Update(ctx, updatedNS, metav1.UpdateOptions{}); err != nil {
-				return fmt.Errorf("failed to update namespace '%s' with label %s, %v", ns.Name, namespaceVersionLabel, err)
+				// Add `openshift-pipelines.tekton.dev/namespace-reconcile-version` label to namespace
+				// so that rbac won't loop on it again
+				nsLabels[namespaceVersionLabel] = r.version
+				updatedNS.SetLabels(nsLabels)
+
+				// Update the namespace with set labels
+				if _, err = r.kubeClientSet.CoreV1().Namespaces().Update(ctx, updatedNS, metav1.UpdateOptions{}); err != nil {
```

Comment:
Just update the Ns, no need for unreconciledNamespaces, especially we are using the same mutex for different operation, so better avoid this

Reply:
removed all references to unreconciledNamespaces and used `Patch` instead of `Update`